src/frontend/src/Components/map/dataLoaders/advisories.js

```diff
@@ -1,25 +1,15 @@
+import { compareRoutePoints } from '../helpers/spatial';
 import { getAdvisories } from '../../data/advisories';
-import * as slices from '../../../slices';
-import * as helpers from '../helpers';
 
-export const loadAdvisories = async (route, advisories, filteredAdvisories, advisoryFilterPoints, dispatch, displayError) => {
+export const loadAdvisories = async (route, advisories, filteredAdvisories, advisoryFilterPoints, dispatch, displayError, worker) => {
   const routePoints = route ? route.points : null;
 
-  // Load if filtered advisories don't exist or route doesn't match
-  if (!filteredAdvisories || !helpers.compareRoutePoints(routePoints, advisoryFilterPoints)) {
+  // Load if filtered objs don't exist or route doesn't match
+  if (!filteredAdvisories || !compareRoutePoints(routePoints, advisoryFilterPoints)) {
     // Fetch data if it doesn't already exist
     const advisoryData = advisories ? advisories : await getAdvisories().catch((error) => displayError(error));
 
-    // Filter data by route
-    const filteredAdvisoryData = route ? helpers.filterAdvisoryByRoute(advisoryData, route, null, true) : advisoryData;
-
-    dispatch(
-      slices.updateAdvisories({
-        list: await getAdvisories().catch((error) => displayError(error)),
-        filteredList: filteredAdvisoryData,
-        filterPoints: route ? route.points : null,
-        timeStamp: new Date().getTime(),
-      }),
-    );
+    // Trigger filter worker
+    worker.postMessage(JSON.stringify({data: advisoryData, route: (route && route.routeFound ? route : null), action: 'updateAdvisories'}));
```

Comment:
Is it necessary to stringify the data structure?  .postMessage appears to handle serialization/deserialization for any argument there, using the structured clone algorithm.  Straightforward JS data objects from parsed JSON output should be fine.